### DISCUSSION TITLE
sync readme with commandlist documentation

### DISCUSSION THREAD

**Discussion by mofosyne** [casey/just#2030] [Ideas]
Maybe have `<!-- just doc start -->` start and end tags and replace it with an autogenerated document of all available commands?

**Comment by casey** [comment]
I'm not sure what you mean. Can you elaborate? Which autogenerated document?

**Comment by mofosyne** [comment]
I'm thinking of two possibilities or simultaneous:

## MINI (Just update a section of a markdown readme doc)

Allow Justfile to search though an existing markdown (or html document) and locate specific 'justfile' tag that allows just to do a replace operation on it. The replaced section is similar to when calling `just --list` but in a more detailed manner (e.g. expected arguments and examples)

| recipes | description  |
| ---          | ---                       |
| [build](JUSTDOC.md#build)    | build stuff        |
| [test](JUSTDOC.md#test)       | test stuff           |

Plus you could have each `[build](JUSTDOC.md#build)` recipes point to a JUSTDOC.md (if full documentation is enabled)

## FULL (Create an entire autogenerated documentation page with all the recipes and how to use it)

Full autogenerated markdown documentation of every command that can be invoked by the user (Maybe with categories like, maintainers, developers, user etc...) based on the comments
    - This would also provide the opportunity to creating a visual graph of dependencies of each recipes, so readers can at a glance see what each recipes will also trigger... (e.g. build will call 'environment_setup' and 'compile' and 'package'). (You might want to allow allow for marking some form of hashtag, so that we show only the most relevant connections. e.g. #buildsystem should not show the logger module, only connection to compile and other relevant recipes)
    - Maybe clicking on a dependency module would also jump to that relevant recipes in the doc

```mermaid
graph TD;
    build-->environment_setup;
    build-->compile-->test;
    build-->package;
```

----

Obviously when I written the initial post, it was just a half baked idea... but hopefully this elaboration will spark further thoughts if this is a good idea or not @casey

**Reply by casey** [reply]
Generating a markdown document with all the commands would be super easy, so I wouldn't be opposed to it, but I would want to get some feedback that this was something that multiple users wanted. Could you create an issue? It's likely to get more comments as an issue, I think.

**Reply by mofosyne** [reply]
Okay thanks https://github.com/casey/just/issues/2033 created

**Reply by goyalyashpal** [reply]
so, this discussion should be closed?

**Reply by mofosyne** [reply]
closed. Hopefully we can get more progress on this idea in the ticket @goyalyashpal
